### ISSUE TITLE
pistachio: build errors with gcc 12

### ISSUE THREAD

**Issue by cproc** [genodelabs/genode#4837]
When building `KERNEL=pistachio BOARD=pc core` with gcc 12, the following errors occur:

```
/.../x86_64-pc-elf/bin/ld: ../../lib/libio.a(print.o): in function `__l4_vsnprintf':
/.../contrib/pistachio-2961c41705536e84699829f8c1d3825c24c063d3/src/kernel/pistachio/user/lib/io/lib.h:39: undefined reference to `strlen'
/.../x86_64-pc-elf/bin/ld: ../../lib/libio.a(print.o): in function `__l4_puts':
/.../contrib/pistachio-2961c41705536e84699829f8c1d3825c24c063d3/src/kernel/pistachio/user/lib/io/lib.h:39: undefined reference to `strlen'
```

**Comment by cproc** [comment]
Fixed by b33c619.

**Comment by cproc** [comment]
Another error:

```
/.../contrib/pistachio-2961c41705536e84699829f8c1d3825c24c063d3/src/kernel/pistachio/kernel/src/api/v4/thread.cc:57:14: error: conflicting declaration 'const tcb_t* __dummy_tcb'
   57 | const tcb_t *__dummy_tcb = (const tcb_t *) &__whole_dummy_tcb;
      |              ^~~~~~~~~~~
In file included from /.../contrib/pistachio-2961c41705536e84699829f8c1d3825c24c063d3/src/kernel/pistachio/kernel/src/api/v4/thread.cc:35:
/.../contrib/pistachio-2961c41705536e84699829f8c1d3825c24c063d3/src/kernel/pistachio/kernel/src/api/v4/tcb.h:772:19: note: previous declaration as 'tcb_t* __dummy_tcb'
  772 |     extern tcb_t *__dummy_tcb;
      |                   ^~~~~~~~~~~
```

Fixed by c85b39b.
